src/main/java/seedu/address/model/task/Task.java

```diff
@@ -96,7 +96,7 @@ public boolean equals(Object other) {
 
         Task otherEvent = (Task) other;
 
-        return associatedEvent.isSameEvent(otherEvent.associatedEvent)
+        return associatedEventName.equals(otherEvent.getAssociatedEventName())
```

Comment:
May I check why we want to check EventName only instead of Event here?

Reply:
I thought of just checking for `eventName` as it is unique for every event, and we use the `eventName` to check for equality under Event::isSameEvent.